### PR TITLE
fix: package import

### PR DESCRIPTION
**Pull request by lianafinaritra**

### CODE REVIEW

File: src/main/java/school/hei/patrimoine/compiler/CasFileCompiler.java [hunk 1]
```diff
@@ -21,7 +22,11 @@ public class CasFileCompiler implements Function<String, Class<?>> {
   }
 
   private static String getClassNameFromPath(String filePath) {
-    return Path.of(filePath).getFileName().toString().replace(".java", "");
+    var path = Path.of(filePath);
+    String fileName = path.getFileName().toString();
+    String className = fileName.substring(0, fileName.lastIndexOf('.'));
+
```
Comment (Mahefaa) [inline src/main/java/school/hei/patrimoine/compiler/CasFileCompiler.java:28]: useless blank space

File: src/main/java/school/hei/patrimoine/compiler/PatrimoineFileCompiler.java [hunk 2]
```diff
@@ -47,7 +48,9 @@ private Class<?> loadClass(String className, Path ioDirPath) {
   private static String getClassNameFromPath(String filePath) {
     var path = Path.of(filePath);
     String fileName = path.getFileName().toString();
-    return fileName.substring(0, fileName.lastIndexOf('.'));
+    String className = fileName.substring(0, fileName.lastIndexOf('.'));
+
```
Comment (Mahefaa) [inline src/main/java/school/hei/patrimoine/compiler/PatrimoineFileCompiler.java:52]: useless blank space

File: src/test/resources/PatrimoineRicheSupplier.java [hunk 3]
```diff
@@ -1,3 +1,5 @@
+package school.hei.patrimoine.cas.drive;
+
```
Comment (Mahefaa) [inline src/test/resources/PatrimoineRicheSupplier.java:2]: could you explain the use of setting the package name ?

test with a different package name also, to ensure that your code works if user's classes are from different packages

File: src/main/java/school/hei/patrimoine/compiler/CasFileCompiler.java (hunk 1 above)
Comment (lianafinaritra) [reply src/main/java/school/hei/patrimoine/compiler/CasFileCompiler.java:28]: done

File: src/main/java/school/hei/patrimoine/compiler/PatrimoineFileCompiler.java (hunk 2 above)
Comment (lianafinaritra) [reply src/main/java/school/hei/patrimoine/compiler/PatrimoineFileCompiler.java:52]: done

File: src/test/resources/PatrimoineRicheSupplier.java (hunk 3 above)
Comment (lianafinaritra) [reply src/test/resources/PatrimoineRicheSupplier.java:2]: done

File: src/test/java/school/hei/patrimoine/compiler/PackageNameExtractorTest.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+
+class PackageNameExtractorTest {
+
+  @SneakyThrows
```
[the remark this answers is not in the document]
Comment (Mahefaa) [reply src/test/java/school/hei/patrimoine/compiler/PackageNameExtractorTest.java:13]: no need for sneaky throws for tests

File: src/test/java/school/hei/patrimoine/compiler/PackageNameExtractorTest.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  @Test
+  void extract_class_name_from_string_ok() {
+    ClassLoader classLoader = PackageNameExtractorTest.class.getClassLoader();
+    URL resourceUrl = classLoader.getResource("PatrimoineRicheSupplier.java");
+    Path resourcePath = Paths.get(resourceUrl.toURI());
+
+    if (resourceUrl == null) {
+      throw new IllegalStateException(
```
Comment (Mahefaa) [inline src/test/java/school/hei/patrimoine/compiler/PackageNameExtractorTest.java:21]: RuntimeException

File: src/main/java/school/hei/patrimoine/compiler/PackageNameExtractor.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+      String line;
+      while ((line = reader.readLine()) != null) {
+        line = line.trim();
+        if (line.startsWith("package ")) {
+          return line.substring(8, line.indexOf(";")).trim();
+        }
+      }
+    }
+    return "";
```
[the remark this answers is not in the document]
Comment (Mahefaa) [reply src/main/java/school/hei/patrimoine/compiler/PackageNameExtractor.java:24]: return optional or return string but throw runtime if none found

File: src/main/java/school/hei/patrimoine/compiler/PackageNameExtractor.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+  @SneakyThrows
+  @Override
+  public String apply(String filePath) {
+    File file = new File(filePath);
+    try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+      String line;
+      while ((line = reader.readLine()) != null) {
+        line = line.trim();
+        if (line.startsWith("package ")) {
```
Comment (Mahefaa) [inline src/main/java/school/hei/patrimoine/compiler/PackageNameExtractor.java:19]: store "package " in a variable and substring from its length

File: src/main/java/school/hei/patrimoine/compiler/CasFileCompiler.java [hunk 8]
```diff
@@ -21,7 +21,12 @@ public class CasFileCompiler implements Function<String, Class<?>> {
   }
 
   private static String getClassNameFromPath(String filePath) {
-    return Path.of(filePath).getFileName().toString().replace(".java", "");
+    PackageNameExtractor packageNameExtractor = new PackageNameExtractor();
```
[the remark this answers is not in the document]
Comment (lianafinaritra) [reply src/main/java/school/hei/patrimoine/compiler/CasFileCompiler.java:24]: done

File: src/main/java/school/hei/patrimoine/compiler/PatrimoineFileCompiler.java [hunk 9]
```diff
@@ -45,9 +45,12 @@ private Class<?> loadClass(String className, Path ioDirPath) {
   }
 
   private static String getClassNameFromPath(String filePath) {
+    PackageNameExtractor packageNameExtractor = new PackageNameExtractor();
```
[the remark this answers is not in the document]
Comment (lianafinaritra) [reply src/main/java/school/hei/patrimoine/compiler/PatrimoineFileCompiler.java:48]: done

File: src/main/java/school/hei/patrimoine/compiler/PackageNameExtractor.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (lianafinaritra) [reply src/main/java/school/hei/patrimoine/compiler/PackageNameExtractor.java:24]: done

File: src/test/java/school/hei/patrimoine/compiler/PackageNameExtractorTest.java (hunk 5 above)
Comment (lianafinaritra) [reply src/test/java/school/hei/patrimoine/compiler/PackageNameExtractorTest.java:21]: done

File: src/main/java/school/hei/patrimoine/compiler/PackageNameExtractor.java (hunk 7 above)
Comment (lianafinaritra) [reply src/main/java/school/hei/patrimoine/compiler/PackageNameExtractor.java:19]: done

File: src/test/java/school/hei/patrimoine/compiler/PackageNameExtractorTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (lianafinaritra) [reply src/test/java/school/hei/patrimoine/compiler/PackageNameExtractorTest.java:13]: done

File: src/main/java/school/hei/patrimoine/compiler/PackageNameExtractor.java [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+import lombok.SneakyThrows;
+
+public class PackageNameExtractor implements Function<String, String> {
+
+  @SneakyThrows
+  @Override
+  public String apply(String filePath) {
+    File file = new File(filePath);
+    try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+      String line;
```
[the remark this answers is not in the document]
Comment (lianafinaritra) [reply src/main/java/school/hei/patrimoine/compiler/PackageNameExtractor.java:16]: done

**Pull request merged**